api/turing/config/config.go

```diff
@@ -94,10 +94,15 @@ func (c *Config) Validate() error {
 
 // BatchEnsemblingConfig captures the config related to the running of batch runners
 type BatchEnsemblingConfig struct {
-	Enabled             bool                `validate:"required"`
-	JobConfig           JobConfig           `validate:"required"`
-	RunnerConfig        RunnerConfig        `validate:"required"`
-	ImageBuildingConfig ImageBuildingConfig `validate:"required"`
+	// Due to the limitations of the go-playground/validator,
```

Comment:
Picking up from [this](https://github.com/gojek/turing/pull/119#discussion_r747988031) comment, I think that all 3 validate tags can be changed to `validate:"required_if=Enabled true"`. As you've highlighted, `Enabled=false` will still evaluate any configs that are set and I believe the `StructFiltered` should be the way around it.

I feel it's not critical to fix, so if it will take some time to address, we could add a TODO here to summarize the plan for the fix and move on for now.

Reply:
Oh ya i changed this already.